---
title: "Reconstructing phylogenies"
author: "January Weiner"
date: "5/27/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

1st. Reading MSA

```{r}
library(phangorn)
library(seqinr)
url <- "https://january3.github.io/Bioinformatics/Practicals/practicals_02_220421/data/hba.aln"
msa <- read.alignment(url, format="clustal")
```

2. Modifying the MSA

```{r}
## names of the sequences are in msa$nam
msa$nam <- gsub(".*_", "", msa$nam)
msa <- as.phyDat(msa, type="AA")
```

3. Doing something useful

```{r}
dist_mat <- dist.ml(msa, model="Blosum62")
mtx <- as.matrix(dist_mat)
heatmap(mtx, symm=TRUE)
```

```{r}
nj_tree <- NJ(dist_mat)
nj_tree <- root(nj_tree, c("GADMO", "POGSC"))
```

Creating a UPGMA tree

```{r eval=FALSE, include=FALSE}
upgma_tree <- upgma(dist_mat)
plot(upgma_tree)
```


